src/TConsole.cpp

```diff
@@ -344,23 +344,29 @@ TConsole::TConsole(Host* pH, ConsoleType type, QWidget* parent)
     mpLineEdit_networkLatency->setContentsMargins(0, 0, 0, 0);
     mpLineEdit_networkLatency->setAlignment(Qt::AlignHCenter | Qt::AlignVCenter);
 
-    QFont latencyFont = QFont("Bitstream Vera Sans Mono", 10, QFont::Normal);
-    int width;
-    int maxWidth = 120;
-    width = QFontMetrics(latencyFont).boundingRect(QString("N:0.000 S:0.000")).width();
-    if (width < maxWidth) {
-        mpLineEdit_networkLatency->setFont(latencyFont);
-    } else {
-        QFont latencyFont2 = QFont("Bitstream Vera Sans Mono", 9, QFont::Normal);
-        width = QFontMetrics(latencyFont2).boundingRect(QString("N:0.000 S:0.000")).width();
-        if (width < maxWidth) {
-            mpLineEdit_networkLatency->setFont(latencyFont2);
-        } else {
-            QFont latencyFont3 = QFont("Bitstream Vera Sans Mono", 8, QFont::Normal);
-            width = QFontMetrics(latencyFont3).boundingRect(QString("N:0.000 S:0.000")).width();
-            mpLineEdit_networkLatency->setFont(latencyFont3);
-        }
-    }
+    int latencyFontPointSize = 21;
+    const int latencyFontSizeMargin = 10;
+    QString dummyTextA = tr("N:%1 S:%2",
```

Comment:
This is interesting but the new code is 1,800% slower than the old code and it'll run any time a new miniconsole/userwindow is created 🤯 

See the benchmarking code I added [in this branch](https://github.com/Mudlet/Mudlet/tree/add-benchmarking-code).

I think that is a high cost to pay, especially when we are a performance-focused client. What do you think?

Reply:
One thousand and eight hundred percent slower? :confused: Oh, your locale number formatting is different.

The reason for the slow down is because this - like **all** the bottom button code - is stuck in the `TConsole` class whereas it is only used in the `TMainConsole` one! There is a whole load of stuff that doesn't belong here, IMHO, but that is a whole new PR!

Reply:
I think I can put in something to bypass it for everything except the `TConsole::MainConsole` case...

Reply:
No it really is 1800% slower, that's not a decimal point!